src/CodeGen_C.h

```diff
@@ -98,6 +98,9 @@ class CodeGen_C : public IRPrinter {
     /** True if there is a void * __user_context parameter in the arguments. */
     bool have_user_context;
 
+    /** The target we're generating code for */
```

Comment:
Adding Target to CodeGen_C for this purpose is not a great design. Whether something is on stack or not really should be encapsulated in the codegen module itself. For C/C++, one likely wants to make it a parameter with a sensible default value, but all inside of CodeGen_C.